### PR TITLE
feat(codeaction): add SpaceBeforeOpeningBrace quick fix

### PR DESCRIPTION
**Pull request by albertocavalcante**
## Summary

- Add LSP quick fix handler for CodeNarc `SpaceBeforeOpeningBrace` rule
- Add OpenRewrite recipe for batch refactoring via Gradle

## Test plan

- [x] FixHandler tests pass (`FormattingFixHandlersTest.kt`)
- [x] OpenRewrite recipe tests pass (`AddSpaceBeforeOpeningBraceTest.kt`)
- [x] All pre-commit hooks pass (spotless, branch-check, etc.)

---

> [!NOTE]
> Adds automated spacing before `{` across LSP quick fixes and batch refactoring.
> 
> - Registers `SpaceBeforeOpeningBrace` in `FixHandlerRegistry` with `fixSpaceBeforeOpeningBrace` to insert a space before `{` when missing
> - New OpenRewrite recipe `AddSpaceBeforeOpeningBrace` handling classes, methods, control blocks, and trailing closures; wired into `codenarc.yml` via `FixAllCodeNarcIssues`
> - Tests: `FormattingFixHandlersTest.kt` (handler) and `AddSpaceBeforeOpeningBraceTest.kt` (recipe) covering positive, negative, and multiline cases
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit e01c3f7118ac55032e4fdbfb4b5ac3eddaa157cf. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

## Summary by CodeRabbit

* **New Features**
  * Automatic formatting that inserts a space before opening braces across Groovy constructs (methods, closures, conditionals, classes, loops, try/catch, lambdas).
  * New formatting recipe added to the toolchain to apply this rule.

* **Tests**
  * Comprehensive unit and integration tests covering positive cases, edge cases (trailing closures, multiline), and no-op scenarios when spacing is already correct.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: refactor/openrewrite/rewrite-codenarc/src/main/kotlin/com/github/albertocavalcante/refactor/codenarc/formatting/AddSpaceBeforeOpeningBrace.kt [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+            }
+
+            override fun visitLambda(lambda: J.Lambda, ctx: ExecutionContext): J.Lambda {
+                var l = super.visitLambda(lambda, ctx)
+
+                // For Groovy trailing closures (with OmitParentheses marker),
+                // the space goes on the lambda itself, not on the body block
+                val hasOmitMarker =
+                    l.markers.findFirst(org.openrewrite.java.marker.OmitParentheses::class.java).isPresent ||
+                        l.markers.findFirst(org.openrewrite.groovy.marker.OmitParentheses::class.java).isPresent
```
[the remark this answers is not in the document]
Comment (albertocavalcante) [reply refactor/openrewrite/rewrite-codenarc/src/main/kotlin/com/github/albertocavalcante/refactor/codenarc/formatting/AddSpaceBeforeOpeningBrace.kt]: Fixed in bbe594d3. Using idiomatic Kotlin `any` instead of Java-style `findFirst().isPresent`.
